python/pyarrow/parquet.py

```diff
@@ -1006,30 +1092,47 @@ class ParquetDataset:
     Check that individual file schemas are all the same / compatible.
 filters : List[Tuple] or List[List[Tuple]] or None (default)
     List of filters to apply, like ``[[('x', '=', 0), ...], ...]``. This
-    implements partition-level (hive) filtering only, i.e., to prevent the
-    loading of some files of the dataset.
-
-    Predicates are expressed in disjunctive normal form (DNF). This means
-    that the innermost tuple describe a single column predicate. These
-    inner predicate make are all combined with a conjunction (AND) into a
-    larger predicate. The most outer list then combines all filters
-    with a disjunction (OR). By this, we should be able to express all
-    kinds of filters that are possible using boolean logic.
-
-    This function also supports passing in as List[Tuple]. These predicates
-    are evaluated as a conjunction. To express OR in predicates, one must
-    use the (preferred) List[List[Tuple]] notation.
+    implements partition-level (hive) filtering, i.e., to prevent the
```

Comment:
I find the hive mention a bit confusing here. Could you add clearer note that depending on use_legacy_dataset what can the user filter on?

Reply:
@kszucs @bkietz can you check if the latest docstring is clearer? (and thanks @bkietz for the wording suggestions!)

Reply:
:+1: